### DISCUSSION TITLE
Flutter iOS IPA Build: Do we really need a script to upload DSYM file(s)?

### DISCUSSION THREAD

**Discussion by orkun1675** [codemagic-ci-cd/codemagic-docs#2899] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

The guidelines at https://docs.codemagic.io/knowledge-firebase/firebase-crashlytics-dsym-uploading/ recommend adding a publish script that manually locates and uploads DSYM files to Frashlytics.

However, if one follows the steps at https://firebase.google.com/docs/crashlytics/get-started?platform=flutter to setup Crashlytics their `project.pbxproj` file will be updated after running `flutterfire configure` to include the following build step:

```
		F15A6494D061B26C7BA402B5 /* [firebase_crashlytics] Crashlytics Upload Symbols */ = {
			isa = PBXShellScriptBuildPhase;
			buildActionMask = 2147483647;
			files = (
			);
			inputFileListPaths = (
			);
			inputPaths = (
				"\"${DWARF_DSYM_FOLDER_PATH}/${DWARF_DSYM_FILE_NAME}/Contents/Resources/DWARF/${TARGET_NAME}\"",
				"\"$(SRCROOT)/$(BUILT_PRODUCTS_DIR)/$(INFOPLIST_PATH)\"",
			);
			name = "[firebase_crashlytics] Crashlytics Upload Symbols";
			outputFileListPaths = (
			);
			outputPaths = (
			);
			runOnlyForDeploymentPostprocessing = 0;
			shellPath = /bin/sh;
			shellScript = "\"$PODS_ROOT/FirebaseCrashlytics/upload-symbols\" --flutter-project \"$PROJECT_DIR/firebase_app_id_file.json\" ";
		};
```

If I'm reading this correctly, both of these scripts are achieving the same thing with a slight caveat:
* The former seems to upload all DSYM fils under `build/ios/archive/Runner.xarchive/dSYMs`: which contains 63 files for me
* The latter only uploads `build/ios/archive/Runner.xarchive/dSYMs/Runner.app.dSYM/Contents/Resources/DWARF/Runner`

So my question for Codemagic folks is: is the latter script executed during Codemagic builds? Why do we need the former script? Is `flutterfire configure` not uploading sufficient information to begin with?

Thank you for your time.

### Expected results

N/A

### Actual results

N/A

**Comment by dam-ease** [comment]
Hello @orkun1675,
You are very right that both scripts aim to upload debug symbol (DSYM) files to Crashlytics.
When you run `flutterfire configure`, it modifies the Xcode project to include a build phase script that automatically uploads the relevant DSYM files to Firebase Crashlytics during the build process. This script handles the necessary uploads, specifically for the primary DSYM files (e.g., Runner.app.dSYM). Just as you mentioned for additional DSYM files that might need to be uploaded for full coverage, I think the codemagic script could be a form of fallback to accounts for edge case. Hence it might not be necessary to include this. If you’ve confirmed that `flutterfire configure` works perfectly for your DSYM uploads, I think you might not need to add the script. However, if you encounter issues with missing symbols in Crashlytics, consider adding the codemagic script in your workflow.

**Comment by orkun1675** [comment]
Thanks, we'll give it a test without the additional script to see what happens.
